### PR TITLE
Add source dependency of test suite implementation to main function

### PR DESCRIPTION
**Pull request by TeBoring**
This change adds a source dependency of the test suite implementaion
class in the main function. For generality reason, the main function is
moved to the file of the test suite implemetation.  New test suite
implementation will need to implement the main function.
In order to make it easy for test suite implementation to implement the
main function, this change also refactor out the common code out of the
main function.

### CODE REVIEW

**Comment by xfxyjwf** [comment]
Can you do just enough in this PR to make the tests work again and do any follow-up refactoring in a separate PR? Ideally this PR should only contain changes to Makefile.am.

**Comment by TeBoring** [comment]
Done in #5073

**Comment by xfxyjwf** [comment]
Bo, can you update the description of this PR accordingly?

**Comment by TeBoring** [comment]
Done

File: conformance/conformance_test.h [hunk 1]
```diff
@@ -89,28 +139,20 @@ class ConformanceTestRunner {
...
 //
 //    int main() {
-//      MyConformanceTestRunner runner;
-//      const std::set<ConformanceTestSuite*>& test_suite_set =
-//          ::google::protobuf::GetTestSuiteSet();
-//      for (auto suite : test_suite_set) {
-//        suite->RunSuite(&runner, &output);
-//      }
+//      MyConformanceTestSuite suite;
+//      MyConformanceTestRunner::Run(argc, argv, &sutie);
```
Comment (haberman) [inline conformance/conformance_test.h]: check spelling of "suite"

**Pull request merged**